src/session/content/message_row/media_picture.rs

```diff
@@ -162,8 +164,22 @@ impl MediaPicture {
         }
 
         let self_ = imp::MediaPicture::from_instance(self);
+        if let Some(old_paintable) = self_.paintable.borrow().as_ref() {
+            if let Some(id) = self_.handler_id.take() {
```

Comment:
In this case the inner `if let` statement is obsolete. The `handler_id` must never be `None` if the paintable is `Some`.
It can be directly unwrapped.